cs-mail/src/main/java/io/cloudslang/content/mail/utils/MessageStoreUtils.java

```diff
@@ -0,0 +1,187 @@
+        boolean useClientCert = false;
+        boolean useTrustCert = false;
+
+        String separator = System.getProperty(PropNames.FILE_SEPARATOR);
+        String javaKeystore = System.getProperty(PropNames.JAVA_HOME) + separator + "lib" + separator + "security" + separator + "cacerts";
+
+        if (keystore.length() == 0 && !trustAllRoots) {
+            boolean storeExists = new File(javaKeystore).exists();
+            keystore = (storeExists) ? Constants.FILE + javaKeystore : null;
+            if (null != keystorePassword) {
+                if (StringUtils.EMPTY.equals(keystorePassword)) {
+                    keystorePassword = SecurityConstants.DEFAULT_PASSWORD_FOR_STORE;
+                }
+            }
+            useClientCert = storeExists;
+        } else {
+            if (!trustAllRoots) {
+                if (!keystore.startsWith(Constants.HTTP))
+                    keystore = Constants.FILE + keystore;
+                useClientCert = true;
+            }
+        }
+
+        if (trustKeystore.length() == 0 && !trustAllRoots) {
+            boolean storeExists = new File(javaKeystore).exists();
+            trustKeystore = (storeExists) ? Constants.FILE + javaKeystore : null;
+            if (storeExists) {
+                if (isEmpty(trustPassword)) {
+                    trustPassword = SecurityConstants.DEFAULT_PASSWORD_FOR_STORE;
+                }
+            } else {
+                trustPassword = null;
+            }
+            useTrustCert = storeExists;
+        } else {
+            if (!trustAllRoots) {
+                if (!trustKeystore.startsWith(Constants.HTTP))
+                    trustKeystore = Constants.FILE + trustKeystore;
+                useTrustCert = true;
+            }
+
+        }
+
+        TrustManager[] trustManagers = null;
+        KeyManager[] keyManagers = null;
+
+        if (trustAllRoots) {
+            trustManagers = new TrustManager[]{new EasyX509TrustManager()};
+        }
+
+        if (useTrustCert) {
+            KeyStore trustKeyStore = SSLUtils.createKeyStore(new URL(trustKeystore), trustPassword);
+            trustManagers = SSLUtils.createAuthTrustManagers(trustKeyStore);
+        }
+        if (useClientCert) {
+            KeyStore clientKeyStore = SSLUtils.createKeyStore(new URL(keystore), keystorePassword);
+            keyManagers = SSLUtils.createKeyManagers(clientKeyStore, keystorePassword);
+        }
+
+        SSLContext context = SSLContext.getInstance(SecurityConstants.SSL);
+        context.init(keyManagers, trustManagers, new SecureRandom());
+        SSLContext.setDefault(context);
+    }
+
+
+    public static Store tryTLSOtherwiseTrySSL(Authenticator auth, Properties props, GetMailInput input) throws MessagingException {
+        Store store = configureStoreWithTLS(props, auth, input);
+        try {
+            store.connect(input.getHostname(), input.getUsername(), input.getPassword());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code contains a potential information leakage in the error handling for the following call(s): &#39;GetMailInput.getPassword()&#39; (line 152). You are handling this error data through: &#39;Store.connect()&#39; (line 152) [with catch clause(s) at: [line 153]]. Please use your organization-specific sensitive information handling loggers or redaction. [Learn more about information leakage](https://cwe.mitre.org/data/definitions/209.html) (external link).